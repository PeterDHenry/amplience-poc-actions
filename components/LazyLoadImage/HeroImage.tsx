import React from 'react';
import { SimpleImg } from 'react-simple-img';

function HeroImage( props ) {
  const imageUrl = `https://cdn.media.amplience.net/i/wigglecrcnonprd/${props.id}`;
  return (
    <SimpleImg
      src={imageUrl}
      animationDuration={1}
      width={1920}
      height={650}
      applyAspectRatio={true}
    />
  );
}

HeroImage.defaultProps = {
  id: "homepage-h-d"
}

export default HeroImage;
