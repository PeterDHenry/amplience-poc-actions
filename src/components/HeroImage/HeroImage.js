import React from 'react';
import { SimpleImg } from 'react-simple-img';

function SpeakerImage({ id }) {
  const imageUrl = `https://cdn.media.amplience.net/i/wigglecrcnonprd/${id}`;
  return (
    <SimpleImg
      src={imageUrl}
      animationDuration="1"
      width={1920}
      height={650}
      applyAspectRatio="true"
    />
  );
}

export default SpeakerImage;
