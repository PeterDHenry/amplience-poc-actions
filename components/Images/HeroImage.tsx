import React from 'react'
import LazyLoadImage from '../Images/LazyLoadImage'

interface HeroImageProps {
  altText: string
  id: string
}

function HeroImage({ id, altText }: HeroImageProps) {
  return (
    <LazyLoadImage
      altText={altText}
      id={id}
      animationDuration={0.1}
      width={1920}
      height={650}
      applyAspectRatio={true}
    />
  )
}

HeroImage.defaultProps = {
  altText: 'Default hero alt text',
  id: 'homepage-h-d',
}

export default HeroImage
