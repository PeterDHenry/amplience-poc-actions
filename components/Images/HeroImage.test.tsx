import { render } from '@testing-library/react'
import { HeroImage } from './HeroImage'

describe('HeroImage component', () => {
  it('matches snapshot', () => {
    //uses default is this correct?
    const { asFragment } = render(<HeroImage />, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
