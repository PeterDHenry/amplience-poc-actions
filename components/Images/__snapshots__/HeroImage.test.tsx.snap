// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage component matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: relative; display: block; padding-bottom: 33.85416666666667%;"
  >
    <noscript />
    <img
      alt="Default hero alt text"
      data-src="https://cdn.media.amplience.net/i/wigglecrcnonprd/homepage-h-d"
      src="data:image/gif;base64,R0lGODlhAQABAAD/ACwAAAAAAQABAAACADs="
      style="height: 100%; width: 100%; visibility: hidden; position: absolute; top: 0px; left: 0px;"
    />
    <div
      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; background: white; transition: 0.1s opacity;"
    />
  </div>
</DocumentFragment>
`;
